libpod/container.go

```diff
@@ -808,3 +814,42 @@ func (c *Container) isStopped() (bool, error) {
 	}
 	return c.state.State == ContainerStateStopped, nil
 }
+
+// nil container state to the database
+func (c *Container) save() error {
+	if err := c.runtime.state.SaveContainer(c); err != nil {
+		return errors.Wrapf(err, "error saving container %s state", c.ID())
+	}
+	return nil
+}
+
+// CleanupStorage unmounts all mount points in container and cleans up container storage
+func (c *Container) CleanupStorage() error {
+	for _, mount := range c.mounts {
```

Comment:
These can probably be moved inside the `if c.state.Mounted {}` block - if we have things mounted but the container itself is not mounted something is very wrong

Reply:
Fine, I was thinking eventally other mount points might get mounted on a place without storage, but I have no problem moving it.

Fixed